crates/credentials/src/vc.rs

```diff
@@ -106,6 +198,42 @@ impl VerifiableCredential {
     }
 }
 
+pub fn validate_vc_data_model(vc: &VerifiableCredential) -> Result<(), CredentialDataModelValidationError> {
```

Comment:
I'm generally more inclined to place this within the `verify()` function since that's the only place it's used, unless you know of a use case where this would need to be called from elsewhere? I know it'll make that function large (in number-of-lines), but that's okay, it's not any less large by splitting it out into a callable function. You could imagine, we expose this function as a callable function for consumers of our SDK and now we're locked in to always supporting it without inducing a breaking change (guiding principle being: be conservative in what is made public)

Open to being wrong on this

Reply:
at a minimum, I would recommend removing the `pub` and keep the function internal to the `vc` rust module

Reply:
ya know I actually don't have strong opinions on this, apologies for the digress 

but I do think it's a good idea to remove the `pub` unless you're aware of a use case where this would be called external to this module